src/models/room.ts

```diff
@@ -309,6 +309,10 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
     private notificationCounts: NotificationCount = {};
     private readonly threadNotifications = new Map<string, NotificationCount>();
     public readonly cachedThreadReadReceipts = new Map<string, CachedReceiptStructure[]>();
+    // Useful to know at what point the current user has started using threads in this room
+    public oldestThreadedReceiptTs = Infinity;
+    // Important to compute `hasUserReadEvent` and similar methods correctly.
+    public unthreadedReceipts = new Map<string, Receipt>();
```

Comment:
wouldn't it be better to make these private and give them accessors, to ensure they can't be modified outside this class?